### PR TITLE
Wwst 2229: Add DTH for gas valve

### PR DESCRIPTION
**Pull request by FenMei007S**
@tpmanley @varzac

### CODE REVIEW

**Comment by FenMei007S** [comment]
@greens  @tpmanley  plz review it

File: devicetypes/smartthings/ezex-gas-valve.src/ezex-gas-valve.groovy [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+        status "off": "on/off: 0"
+
+        // reply messages
+        reply "zcl on-off on": "on/off: 1"
+        reply "zcl on-off off": "on/off: 0"
+    }
+
+    tiles(scale: 2) {
+        multiAttributeTile(name:"contact", type: "generic", width: 6, height: 4, canChangeIcon: true){
+            tileAttribute ("device.contact", key: "PRIMARY_CONTROL") {
```
Comment (greens) [inline devicetypes/smartthings/ezex-gas-valve.src/ezex-gas-valve.groovy]: should be `device.valve`

**Review by greens — CHANGES_REQUESTED** [review]
Just add the fingerprint to `zigbee-valve` as far as I can tell you've just copy-pasted that entire DTH

**Comment by FenMei007S** [comment]
@greens, i know just add fingerprint, if zigbee-valve can meet the request, but this device not report the battery remain precentage, just report battery low and normal, custom ui metadata made to meet this, vid "ezex-gas-valve", and define sendBatteryEvent func.

**Comment by greens** [comment]
@FenMei007S You can still add the fingerprint and the custom battery handling to the zigbee-valve DTH (preferably behind a check of the model name). You can also specify device-specific metadata in the fingerprint now. 

See here for an example: https://github.com/SmartThingsCommunity/SmartThingsPublic/blob/master/devicetypes/smartthings/zll-white-color-temperature-bulb-5000k.src/zll-white-color-temperature-bulb-5000k.groovy

**Comment by FenMei007S** [comment]
@greens  Updated, plz review

File: devicetypes/smartthings/zigbee-valve.src/zigbee-valve.groovy [hunk 2]
```diff
@@ -39,7 +41,7 @@ metadata {
     }
 
     tiles(scale: 2) {
-        multiAttributeTile(name:"contact", type: "generic", width: 6, height: 4, canChangeIcon: true){
+        multiAttributeTile(name:"valve", type: "generic", width: 6, height: 4, canChangeIcon: true){
             tileAttribute ("device.contact", key: "PRIMARY_CONTROL") {
```
Comment (greens) [inline devicetypes/smartthings/zigbee-valve.src/zigbee-valve.groovy]: you'll also want to change this to `device.valve`

File: devicetypes/smartthings/zigbee-valve.src/zigbee-valve.groovy [hunk 3]
```diff
@@ -39,7 +41,7 @@ metadata {
     }
 
     tiles(scale: 2) {
-        multiAttributeTile(name:"contact", type: "generic", width: 6, height: 4, canChangeIcon: true){
+        multiAttributeTile(name:"valve", type: "generic", width: 6, height: 4, canChangeIcon: true){
```
Comment (greens) [inline devicetypes/smartthings/zigbee-valve.src/zigbee-valve.groovy]: Please leave this tile as "contact". There are local execution reasons that this is contact and not "valve".

File: devicetypes/smartthings/zigbee-valve.src/zigbee-valve.groovy [hunk 4]
```diff
@@ -88,38 +90,57 @@ def parse(String description) {
...
-
         sendEvent(event)
     }
     else {
         def descMap = zigbee.parseDescriptionAsMap(description)
         if (descMap.clusterInt == CLUSTER_BASIC && descMap.attrInt == BASIC_ATTR_POWER_SOURCE){
             def value = descMap.value
             if (value == "01" || value == "02") {
-                sendEvent(name: "powerSource", value: "mains")
+                sendEvent(name: "powerSource", value: "Mains")
```
Comment (greens) [inline devicetypes/smartthings/zigbee-valve.src/zigbee-valve.groovy]: All of these values should be left lowercase. The capability and ocf mapping require lowercase.

File: devicetypes/smartthings/zigbee-valve.src/zigbee-valve.groovy [hunk 5]
```diff
@@ -133,7 +154,9 @@ def refresh() {
     zigbee.onOffRefresh() +
     zigbee.readAttribute(CLUSTER_BASIC, BASIC_ATTR_POWER_SOURCE) +
     zigbee.readAttribute(CLUSTER_POWER, POWER_ATTR_BATTERY_PERCENTAGE_REMAINING) +
+    zigbee.readAttribute(zigbee.IAS_ZONE_CLUSTER, zigbee.ATTRIBUTE_IAS_ZONE_STATUS) +
```
Comment (greens) [inline devicetypes/smartthings/zigbee-valve.src/zigbee-valve.groovy]: these commands should only be sent if the model name is `"E253-KR0B0ZX-HA"`

File: devicetypes/smartthings/zigbee-valve.src/zigbee-valve.groovy [hunk 6]
```diff
@@ -88,38 +90,57 @@ def parse(String description) {
...
-                sendEvent(name: "powerSource", value: "unknown")
+                sendEvent(name: "powerSource", value: "Unknown")
             }
         }
         else if (descMap.clusterInt == CLUSTER_POWER && descMap.attrInt == POWER_ATTR_BATTERY_PERCENTAGE_REMAINING) {
             event.name = "battery"
             event.value = Math.round(Integer.parseInt(descMap.value, 16) / 2)
             sendEvent(event)
         }
+        else if (descMap.clusterInt == zigbee.IAS_ZONE_CLUSTER && descMap.attrInt == zigbee.ATTRIBUTE_IAS_ZONE_STATUS) {
```
Comment (greens) [inline devicetypes/smartthings/zigbee-valve.src/zigbee-valve.groovy:117]: I want to confirm that you have tested this yourself and that it works.

**Review by greens — CHANGES_REQUESTED** [review]
Leave the "contact" stuff in the tiles.

Verify that your battery handling works as expected.

Only send IAS configs conditionally.

File: devicetypes/smartthings/zigbee-valve.src/zigbee-valve.groovy (hunk 6 above)
Comment (FenMei007S) [reply devicetypes/smartthings/zigbee-valve.src/zigbee-valve.groovy:117]: Yes, i test it, we will receive this message

**Comment by FenMei007S** [comment]
@greens Plz review again, local tested

File: devicetypes/smartthings/zigbee-valve.src/zigbee-valve.groovy [hunk 7]
```diff
@@ -59,8 +61,8 @@ metadata {
             state "default", label:"", action:"refresh.refresh", icon:"st.secondary.refresh"
         }
 
-        main(["contact"])
-        details(["contact", "battery", "refresh"])
+        main(["valve"])
+        details(["valve", "battery", "refresh"])
```
Comment (greens) [inline devicetypes/smartthings/zigbee-valve.src/zigbee-valve.groovy]: you forgot to revert these changes

**Comment by FenMei007S** [comment]
@greens updated

**Review by greens — APPROVED** [review]
